src/services/user.ts

```diff
@@ -57,3 +57,15 @@ export const getUserProfile = async (): Promise<UserProfile> => {
 
   return response.data.attributes;
 };
+
+export const resetPassword = async (email: string) => {
+  const requestBody = {
+    user: {
+      email,
+    },
+    client_id: process.env.NEXT_PUBLIC_AUTH_CLIENT_ID,
+    client_secret: process.env.NEXT_PUBLIC_AUTH_CLIENT_SECRET,
+  };
+
+  return await post('/passwords', requestBody);
```

Comment:
What about declaring `resetPassword` as a [Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise), so that you would put the `try/catch` inside the resetPassword using the generic `resolve` and `reject` functions.

```suggestion
  try {
    await post('/passwords', requestBody);
  }
  catch {
    return reject();
  }
  
  return resolve();
```

In the component, it would then be much "lighter/easier" to consume:

```
const resetPasswordSucceed = () => {
  setFormSuccess(true);
  setFormErrors([]);
  setFormLoading(false);
};

const resetPasswordRejected = () => {
  setFormErrors(['Something went wrong. Please try again later']);
  setFormLoading(false);
};

resetPassword().then(resetPasswordSucceed, resetPasswordRejected);
```

(as often, I suggested that but I did not test — you might encounter some challenges too)

Reply:
I updated as you commented and adjusted it a bit